### PR TITLE
fix: [M3-8578] - Fix console error on Volume Create - Linode Config selection

### PR DESCRIPTION
**Pull request by cpathipa**
## Description 📝
The console Warning: Cannot update a component (VolumeCreate) while rendering a different component (Unknown), usually occurs when a state update is triggered during the rendering phase of another component. In the `ConfigSelect` component, this issue likely arises from the logic that automatically sets a new value. So, moving that logic to `useEffect` hook avoids triggering state updates during the render phase.

## Target release date 🗓️
9/30

### Reproduction steps
(How to reproduce the issue, if applicable)
- Navigate to Volume Create Flow, Select linode and notice warning in the console.

### Verification steps
(How to verify changes)
- Checkout the branch and run in local
-  Navigate to Volume Create Flow, select linode and observe no console warning.

## As an Author I have considered 🤔

*Check all that apply*

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [ ] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

**Review by hkhalil-akamai — APPROVED** [review]
Verified console error is no longer appearing

File: packages/manager/.changeset/pr-10970-fixed-1726762187663.md
Comment (dwiley-akamai) [inline packages/manager/.changeset/pr-10970-fixed-1726762187663.md:1]: Do we need a changeset? I didn't see the warning in prod, only locally

**Review by dwiley-akamai — APPROVED** [review]
Console error not present on this branch compared to `develop` ✅

File: packages/manager/.changeset/pr-10970-fixed-1726762187663.md
Comment (hkhalil-akamai) [reply packages/manager/.changeset/pr-10970-fixed-1726762187663.md:1]: Good point with the changeset. Warnings are suppressed in prod.

**Pull request merged**